web-service/app.py

```diff
@@ -10,6 +12,36 @@
 def root():
     with open("players.json") as player_file:
         player_json = json.loads(player_file.read())
-        teams, remainder = matchmake(player_json['players'], 6)
+        teams, players = matchmake(player_json['players'], 6)
 
-    return "Teams: %s\nRemainder: %s" % (teams, remainder)
+    with open("persistence.json", "w") as outfile:
```

Comment:
Consider separating some of this logic into its own function to enable easy testing without worrying about the Flask part.